packages/@angular/cli/utilities/schematics.ts

```diff
@@ -0,0 +1,252 @@
+/**
+ * Refer to the angular shematics library to let the dependency validator
+ * know it is used..
+ *
+ * require('@schematics/angular')
+ */
+
+import {
+  Collection,
+  DryRunEvent,
+  DryRunSink,
+  FileSystemSink,
+  FileSystemTree,
+  Schematic,
+  SchematicEngine,
+  Tree
+} from '@angular-devkit/schematics';
+import {
+  FileSystemHost,
+  FileSystemSchematicDesc,
+  NodeModulesEngineHost
+} from '@angular-devkit/schematics/tools';
+import { SchemaClassFactory } from '@ngtools/json-schema';
+import { Observable } from 'rxjs/Observable';
+import * as path from 'path';
+import { green, red, yellow } from 'chalk';
+import { CliConfig } from '../models/config';
+import 'rxjs/add/operator/concatMap';
+import 'rxjs/add/operator/map';
+
+const SilentError = require('silent-error');
+
+
+export function getEngineHost() {
+  const engineHost = new NodeModulesEngineHost();
+  return engineHost;
+}
+
+export function getCollection(collectionName: string): Collection<any, any> {
+  const engineHost = getEngineHost();
+  const engine = new SchematicEngine(engineHost);
+
+  // Add support for schemaJson.
+  engineHost.registerOptionsTransform((schematic: FileSystemSchematicDesc, options: any) => {
+    if (schematic.schema) {
+      const SchemaMetaClass = SchemaClassFactory<any>(schematic.schemaJson!);
+      const schemaClass = new SchemaMetaClass(options);
+      return schemaClass.$$root();
+    }
+    return options;
+  });
+
+  const collection = engine.createCollection(collectionName);
+
+  if (collection === null) {
+    throw new SilentError(`Invalid collection (${collectionName}).`);
+  }
+  return collection;
+}
+
+export function getSchematic(collection: Collection<any, any>,
+                             schematicName: string): Schematic<any, any> {
+  return collection.createSchematic(schematicName);
+}
+
+export interface SchematicOptions {
+  dryRun: boolean;
+  force: boolean;
+  [key: string]: any;
+}
+
+export interface SchematicOutput {
+  modifiedFiles: string[];
+}
+
+export function runSchematic(schematic: Schematic<{}, {}>, workingDirectory: string,
+  appDirectory: string, options: SchematicOptions, projectRoot: string): Promise<SchematicOutput> {
+
+  let modifiedFiles: string[] = [];
+
+  const preppedOptions = prepOptions(schematic, appDirectory, options, projectRoot);
+  const opts = { ...options, ...preppedOptions };
+
+  const host = Observable.of(new FileSystemTree(new FileSystemHost(workingDirectory)));
+
+  const dryRunSink = new DryRunSink(workingDirectory, opts.force);
+  const fsSink = new FileSystemSink(workingDirectory, opts.force);
+
+  let error = false;
+  const loggingQueue: string[] = [];
+
+  dryRunSink.reporter.subscribe((event: DryRunEvent) => {
```

Comment:
We should stick to the current format for logging: 
![image](https://user-images.githubusercontent.com/681969/29096397-9bf1eb8c-7c49-11e7-9308-5ec774532677.png)